components/timezone/src/ixdtf.rs

```diff
@@ -78,116 +78,214 @@ impl UtcOffset {
 
 // ==== TimeZoneInfo methods and traits ====
 
-impl TimeZoneInfo {
-    fn try_from_ixdtf_record(ixdtf_record: &IxdtfParseRecord) -> Result<Self, ParseError> {
-        match ixdtf_record {
+impl TimeZoneInfo<models::Full> {
+    fn try_offset_only_from_ixdtf_record(
+        ixdtf_record: &IxdtfParseRecord,
+    ) -> Result<Self, ParseError> {
+        let (offset, date, time) = match ixdtf_record {
+            // -0800
             IxdtfParseRecord {
                 offset: Some(offset),
                 tz: None,
+                date: Some(date),
+                time: Some(time),
                 ..
-            } => Self::try_from_utc_offset_record(offset),
+            } => (offset, date, time),
+            // [-0800]
             IxdtfParseRecord {
                 offset: None,
-                tz: Some(time_zone_annotation),
-                date,
-                time,
+                tz:
+                    Some(TimeZoneAnnotation {
+                        tz: TimeZoneRecord::Offset(offset),
+                        ..
+                    }),
+                date: Some(date),
+                time: Some(time),
                 ..
-            } => Self::try_from_time_zone_record(&time_zone_annotation.tz, None, date, time),
+            } => (offset, date, time),
```

Comment:
these cases can be collapsed

Reply:
no? they cover different cases which are in comments